test/codec.ts

```diff
@@ -75,6 +75,14 @@ describe('codec', () => {
         assert.strictEqual(json, '1986-03-22');
       });
 
+      it('should accept year zero in y/m/d number values', () => {
+        const d = new codec.SpannerDate(null!);
+        const date = new codec.SpannerDate(0, 2, 22);
```

Comment:
why is `(0, 2, 22)` instead of `(0, 3, 22)`? did this test pass? `json` should not be equal to `1900-03-22`.

Reply:
The month are 0 based index ([link](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/getMonth/))
0 = January

Reply:
Thanks. Got it. The built-in `Date` uses zero-based index. This is awful, 😭.